manifests/base/application-controller-deployment/argocd-application-controller-deployment.yaml

```diff
@@ -20,8 +20,8 @@ spec:
       - args:
         - /usr/local/bin/argocd-application-controller
         env:
-        - name: ARGOCD_CONTROLLER_REPLICAS
-          value: "1"
+        - name: ARGOCD_ENABLE_DYNAMIC_CLUSTER_DISTRIBUTION
+          value: "true"
```

Comment:
Not sure if this should be enabled by default, also probably better to pick this up from configmap o keep it uniform with other settings.

Reply:
Regarding the comments here my question is: Does it makes sense to switch the flag `ARGOCD_ENABLE_DYNAMIC_CLUSTER_DISTRIBUTION` to false with using the `Deployment` controller?

- If the answer is yes then I agree with @gdsoumya that this should be defined in the `argocd-cmd-params-cm.yaml` file.
- If the answer is no then I agree with @morey-tech that this doesn't need to be in the configmap. However if this is the case, I don't even understand why we have this flag in the first place

Reply:
As @morey-tech mentioned, the flag is needed only for `application-controller Deployment` to enable the feature  and not for StatefulSets. The flag was added to not execute computations for deployments while using `StatefulSet`.

I do not see a benefit to move this to configMap. Because if we do move to configMap, we would need to add a patch for migrating to StatefulSets as well.

Reply:
Running as a deployment and setting this flag to true can be mutually exclusive imo. As a user I can use the deployment manifest version without switching on dynamic distribution? I feel this should be more of a manual change to enable this or kustomization provided by the user at the time of installation rather than a default.

Reply:
As discussed with @ishitasequeira, we reached an agreement to simply remove the ARGOCD_ENABLE_DYNAMIC_CLUSTER_DISTRIBUTION flag and the dynamic cluster distribution logic will be driven by deploying the controller as a deployment.